clustered/client/src/test/java/org/ehcache/clustered/client/internal/UnitTestConnectionService.java

```diff
@@ -17,90 +17,384 @@
-
-    newServer.start();
-    return newServer;
+  private static final Map<URI, ServerDescriptor> servers = new HashMap<URI, ServerDescriptor>();
+
+  /**
+   * Adds a {@link PassthroughServer} if, and only if, a mapping for the {@code URI} supplied does not
+   * already exist.  The server is started as it is added.
+   *
+   * @param uri the {@code URI} for the server; only the <i>scheme</i>, <i>host</i>, and <i>port</i>
+   *            contribute to the server identification
+   * @param server the {@code PassthroughServer} instance to use for connections to {@code uri}
+   */
+  public static void add(URI uri, PassthroughServer server) {
+    URI keyURI = createKey(uri);
+    if (servers.containsKey(keyURI)) {
+      throw new AssertionError("Server at " + uri + " already provided; use remove() to remove");
+    }
+
+    servers.put(keyURI, new ServerDescriptor(server));
+    server.start();
+    LOGGER.info("Started PassthroughServer at {}", keyURI);
   }
 
-  private static ResourceType defineResource(String name, MemoryUnit unit, long size) {
-    final ResourceType resource = new ResourceType();
-    resource.setName(name);
-    resource.setUnit(unit);
-    resource.setValue(BigInteger.valueOf(size));
-    return resource;
+  /**
+   * Adds a {@link PassthroughServer} if, and only if, a mapping for the URI supplied does not
+   * already exist.  The server is started as it is added.
+   *
+   * @param uri the URI, in string form, for the server; only the <i>scheme</i>, <i>host</i>, and <i>port</i>
+   *            contribute to the server identification
+   * @param server the {@code PassthroughServer} instance to use for connections to {@code uri}
+   */
+  public static void add(String uri, PassthroughServer server) {
+    add(URI.create(uri), server);
   }
 
-  private static PassthroughServer server;
-  static {
-    try {
-      server = createServer();
-    } catch (Exception ex) {
-      throw new AssertionError(ex);
+  /**
+   * Removes the {@link PassthroughServer} previously associated with the {@code URI} provided.  The
+   * server is stopped as it is removed.  In addition to stopping the server, all open {@link Connection}
+   * instances created through the {@link #connect(URI, Properties)} method are closed; this is done to
+   * clean up the threads started in support of each open connection.
+   *
+   * @param uri the {@code URI} for which the server is removed
+   *
+   * @return the removed {@code PassthroughServer}
+   */
+  public static PassthroughServer remove(URI uri) {
+    URI keyURI = createKey(uri);
+    ServerDescriptor serverDescriptor = servers.remove(keyURI);
+    if (serverDescriptor != null) {
+      serverDescriptor.server.stop();
+      LOGGER.info("Stopped PassthroughServer at {}", keyURI);
+
+      for (Connection connection : serverDescriptor.getConnections().keySet()) {
+        try {
+          LOGGER.warn("Force close {}", formatConnectionId(connection));
```

Comment:
If this were a real server the connections would get torn down from the server end though... shouldn't the passthrough server behave the same way?